vars/simpleMavenLibPipeline.groovy

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/groovy
+
+// TODO: Is body needed or can it be removed?
+def call(Map args, Closure body) {
+  String dockerBuildImage = args["dockerBuildImage"]
+    ?: { throw new RuntimeException("Missing arg: dockerBuildImage") }()
+
+  String dockerNodeLabel = args["dockerNodeLabel"] ?: "docker"
+
+  echo "${dockerBuildImage}"
+  echo "${dockerNodeLabel}"
+
+  dockerNode([label: dockerNodeLabel]) {
+    def buildImage = docker.image(dockerBuildImage)
+    buildImage.pull() // Ensure latest version
+
+    buildImage.inside {
+      stage('Checkout source') {
+        checkout scm
+      }
+
+      try {
+        stage('Build and conditionally release') {
+          withMavenDeployVersionByTimeEnv { String revision ->
+            String goal = env.BRANCH_NAME == "master" && changedSinceLatestTag()
+              ? "source:jar deploy scm:tag"
+              : "verify"
+            sh """
+                        mvn -s \$MAVEN_SETTINGS org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M3:enforce -Drules=requireReleaseDeps
+                        mvn -s \$MAVEN_SETTINGS -B -Dtag=$revision -Drevision=$revision $goal
+                    """
+          }
+        }
+      } finally {
+        // TODO: Needed?
+        // saveJunitReport()
+        // saveJacocoReport()
+      }
+    }
+
+    // TODO: What is this for?
```

Comment:
Kun relevant for "services" der vi bygger et docker image som artifakt. Ikke relevant for libs